ccdproc/bitmask.py

```diff
@@ -0,0 +1,473 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import collections
+import itertools
+
+import numpy as np
+
+_INTEGER_DTYPE_KINDS = {'u', 'i'}
+
+# Powers of two 2**0 - 2**63
+_POWERS = (
+    1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384,
+    32768, 65536, 131072, 262144, 524288, 1048576, 2097152, 4194304, 8388608,
+    16777216, 33554432, 67108864, 134217728, 268435456, 536870912, 1073741824,
+    2147483648, 4294967296, 8589934592, 17179869184, 34359738368, 68719476736,
+    137438953472, 274877906944, 549755813888, 1099511627776, 2199023255552,
+    4398046511104, 8796093022208, 17592186044416, 35184372088832,
+    70368744177664, 140737488355328, 281474976710656, 562949953421312,
+    1125899906842624, 2251799813685248, 4503599627370496, 9007199254740992,
+    18014398509481984, 36028797018963968, 72057594037927936,
+    144115188075855872, 288230376151711744, 576460752303423488,
+    1152921504606846976, 2305843009213693952, 4611686018427387904,
+    9223372036854775808
+)
+
+# Use signed integers for more controlled behaviour expect for 64 layers then
+# only an unsigned may hold the result.
+_REPRESENTABLE_FLAGS = {
```

Comment:
I personally prefer using unsigned int for flag-type variables (maybe because this is what I used to use in C/C++ - see, e.g., DWORD type often returned by many Windows system calls is an unsigned integer: https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751(v=vs.85).aspx).

In addition, ``_REPRESENTABLE_FLAGS`` probably can be determined programmatically. Also, you say:
> Use signed integers for more controlled behaviour

I would think that using unsigned would be more consistent.

Finally, it is strange that 2 byte int16 can represent only 15 bits... Something is not right here.